src/main/webapp/app/entities/programming-exercise/programming-exercise-update.component.ts

```diff
@@ -137,4 +134,40 @@ export class ProgrammingExerciseUpdateComponent implements OnInit {
     trackCourseById(index: number, item: Course) {
         return item.id;
     }
+
+    /**
+     * Change the selected programming language for the current exercise. If there are unsaved changes, the user
+     * will see a confirmation dialog about switching to a new template
+     *
+     * @param language The new programming language
+     * @param languageSelect The <select> HTML element, which caused the language change
+     */
+    onNewProgrammingLanguage(language: ProgrammingLanguage, languageSelect?: HTMLSelectElement) {
```

Comment:
The only way the select element actually reverted back to the old language was by passing it here and calling `.selectedIndex` directly. `[selectedIndex]` in the HTML didn't work